internal/helm/helm.go

```diff
@@ -351,18 +356,26 @@ func MergeVals(
 // PrintHelmTemplateCommand will log a message so that users can replicate
 // the same behavior as the CLI. The log message will be slightly different
 // depending on if 'helmChartDirectory' is set or not.
+// If 'apiVersions' is given, said values will be added to the log message.
 func PrintHelmTemplateCommand(
 	logger utils.Logger,
 	helmValues map[string]interface{},
 	helmChartDirectory string,
 	namespace string,
 	ciliumVer semver2.Version,
+	apiVersions []string,
 ) {
 	valsStr := valuesToString("", helmValues)
+	apiVersionsStr := ""
+	if len(apiVersions) > 0 {
+		for _, av := range apiVersions {
+			apiVersionsStr = fmt.Sprintf("%s --api-versions %s", apiVersionsStr, av)
```

Comment:
I think --api-versions takes a string array, could we do that instead of adding a bunch of individual `--api-versions` flags to make the rendered command cleaner?

Reply:
I tried to see if we could, but I don't think we can. You're right that helm uses a [StringArrayVarP](https://pkg.go.dev/github.com/spf13/pflag#StringArrayVarP) for `--api-versions` ([template.go#L184](https://github.com/helm/helm/blob/main/cmd/helm/template.go#L184)), but any value passed to a `StringArrayVarP` will be taken "as-is" and not be separated by commas. That functionality is implemented in `StringSliceVarP`:

```
# https://pkg.go.dev/github.com/spf13/pflag#StringArrayVarP
StringArrayVar defines a string flag with specified name, default value, and usage string.
The argument p points to a []string variable in which to store the value of the flag.
The value of each argument will not try to be separated by comma. Use a StringSlice for that.
```

I don't think helm does any work to take the values from `--api-versions` and look for CSVs, as the string slice from `--api-versions` is converted to a [VersionSet](https://pkg.go.dev/helm.sh/helm/v3@v3.10.1/pkg/chartutil#VersionSet), which is just an alias for a string slice. It as a method [`Has(apiVersion string) bool`](https://github.com/helm/helm/blob/v3.10.1/pkg/chartutil/capabilities.go#L104) which is just a simple for-loop search through the slice.

I submitted an issue on this upstream, maybe in the future we can revisit: https://github.com/helm/helm/issues/11485